### PR TITLE
docs: fix coverage badge primarily

### PR DESCRIPTION
**Pull request by desenna**
Signed-off-by: Michael de Senna <desenna@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 [![Build Status](https://github.com/buildpacks/lifecycle/workflows/build/badge.svg)](https://github.com/buildpacks/lifecycle/actions)
 [![GoDoc](https://godoc.org/github.com/buildpacks/lifecycle?status.svg)](https://godoc.org/github.com/buildpacks/lifecycle)
-[![codecov](https://codecov.io/gh/buildpacks/pack/branch/main/graph/badge.svg)](https://codecov.io/gh/buildpacks/pack)
+[![codecov](https://codecov.io/gh/buildpacks/lifecycle/branch/main/graph/badge.svg)](https://codecov.io/gh/buildpacks/lifecycle/tree/main)
```
Comment (desenna) [inline README.md:5]: This is to correct the badge. I think though that if someone who has admin permissions to codecov.io could set the default branch to main, as in buildpacks/pack, we could probably drop tree/main.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -87,8 +88,15 @@ jobs:
...
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          file: ./out/tests/coverage-unit.txt
+          flags: unit,os_windows
+          fail_ci_if_error: true
+          verbose: true
       - name: Prepare Codecov
-        uses: crazy-max/ghaction-chocolatey@v1
+        uses: crazy-max/ghaction-chocolatey@v2
```
Comment (desenna) [inline .github/workflows/build.yml:99]: bump to v2 (mirroring what's in buildpacks/pack)

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -29,12 +29,13 @@ jobs:
         env:
           TEST_COVERAGE: 1
         run: make test
-      - name: Upload Coverage
-        uses: codecov/codecov-action@v1.0.5
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```
Comment (desenna) [inline .github/workflows/build.yml:33]: bump to v3 (mirroring what's in buildpacks/pack)

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -29,12 +29,13 @@ jobs:
         env:
           TEST_COVERAGE: 1
         run: make test
-      - name: Upload Coverage
-        uses: codecov/codecov-action@v1.0.5
+      - name: Upload coverage to Codecov
```
Comment (desenna) [inline .github/workflows/build.yml:32]: This language is consistent with that of buildpacks/pack

File: .github/workflows/build.yml (hunk 2 above)
Comment (desenna) [reply .github/workflows/build.yml:92]: TBH I found the split here between linux and windows a bit confusing or I guess I don't really understand it. I'm wondering if only windows installs codecov, both run tests (maybe the linux one is ignored bc not installed?), and only the linux section uploads the coverage-unit.txt to hosted codecov..?

Or is perhaps something implicit happening here that I'm missing?

Comment (natalieparellano) [reply .github/workflows/build.yml:92]: I'm a bit fuzzy on this myself TBH... I do see output from Codecov in the linux workflows, but I'm not sure why it doesn't need to be installed explicitly.

**Review by natalieparellano — APPROVED** [review]

**Comment by natalieparellano** [comment]
Thanks for the PR @desenna!!!

**Pull request merged**